server/subscription.go

```diff
@@ -0,0 +1,62 @@
+package server
+
+import (
+	"net/http"
+
+	enterpriseAPI "github.com/bytebase/bytebase/enterprise/api"
+	"github.com/google/jsonapi"
+	"github.com/labstack/echo/v4"
+)
+
+func (s *Server) registerSubscriptionRoutes(g *echo.Group) {
+	g.GET("/subscription", func(c echo.Context) error {
+		subscription, err := s.loadSubscription()
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to load subscription").SetInternal(err)
+		}
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		if err := jsonapi.MarshalPayload(c.Response().Writer, subscription); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to marshal subscription response").SetInternal(err)
+		}
+		return nil
+	})
+
+	g.PATCH("/subscription", func(c echo.Context) error {
+		patch := &enterpriseAPI.SubscriptionPatch{}
+		if err := jsonapi.UnmarshalPayload(c.Request().Body, patch); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformatted create subscription request").SetInternal(err)
+		}
+
+		if err := s.LicenseService.StoreLicense(patch.License); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create license").SetInternal(err)
+		}
+
+		subscription, err := s.loadSubscription()
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to load subscription").SetInternal(err)
+		}
+
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
+		if err := jsonapi.MarshalPayload(c.Response().Writer, subscription); err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to marshal subscription response").SetInternal(err)
+		}
+		return nil
+	})
+}
+
+func (s *Server) loadSubscription() (*enterpriseAPI.Subscription, error) {
+	license, err := s.LicenseService.LoadLicense()
+	if err != nil {
+		return nil, err
+	}
+
+	s.license = license
```

Comment:
LoadLicense seem to already doing this

Reply:
But the LoadLicense in start up is only used to init license in memory, thus we can set up another middleware to guard specific endpoints. The code here means to refresh the license in memory.

Reply:
Seems I misunderstand your comment. It's duplicate code, I'll remove one

Reply:
Updated. We will load the license on service start up. and refresh license after patch.